fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/steps/WebDriverBasedUserInfoFintech.java

```diff
@@ -0,0 +1,157 @@
+package de.adorsys.fintech.tests.e2e.steps;
+
+import com.tngtech.jgiven.Stage;
+import com.tngtech.jgiven.integration.spring.JGivenStage;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.retry.RetryOperations;
+
+import java.net.URI;
+import java.time.Duration;
+
+import static de.adorsys.fintech.tests.e2e.config.RetryableConfig.TEST_RETRY_OPS;
+import static de.adorsys.fintech.tests.e2e.steps.FintechStagesUtils.*;
+
+@JGivenStage
+@SuppressWarnings("checkstyle:MethodName") // Jgiven prettifies snake-case names not camelCase
+public class WebDriverBasedUserInfoFintech<SELF extends WebDriverBasedUserInfoFintech<SELF>> extends Stage<SELF> {
+
+    static final String FINTECH_URI = "https://obg-dev-fintechui.cloud.adorsys.de";
+
+    @Autowired
+    @Qualifier(TEST_RETRY_OPS)
+    private RetryOperations withRetry;
+
+    @Value("${test.webdriver.timeout}")
+    private Duration timeout;
+
+    public SELF user_opens_fintechui_login_page(WebDriver driver) {
+        driver.get(FINTECH_URI);
+        return self();
+    }
+
+    public SELF user_login_with_its_credentials(WebDriver driver) {
+        sendText(driver, By.id("username"), USERNAME);
+        sendText(driver, By.id("password"), PIN_VALUE);
+        return self();
+    }
+
+    public SELF user_sees_that_does_not_need_to_login(WebDriver webDriver) {
+        waitForPageLoadAndUrlEndsWithPath(webDriver, "search");
+        return self();
+    }
+
+    public SELF user_back_to_bank_search(WebDriver webDriver) {
+        waitForPageLoad(webDriver);
+        clickOnButton(webDriver, By.className("icon__container"), true);
+        return self();
+    }
+
+    public SELF user_wait_for_the_result_in_bank_search(WebDriver webDriver) {
+        wait(webDriver);
+        clickOnButton(webDriver, By.className("bank-list"), false);
```

Comment:
It is very broad element definition - you should add id to the control and use it here